### DISCUSSION TITLE
2 compute passes, write then read from integer texture?

### DISCUSSION THREAD

**Discussion by OneKnightOnly** [gfx-rs/wgpu#2742] [Q&A] [answered]
I'm new to wgpu and trying to wrap my head around layouts and bind groups.  I have a compute shader with writeonly access that writes to an integer texture, how can I then read from that same texture in a second compute shader? do I need to bind a new bind group and layout?

**Comment by cwfitzgerald** [comment, marked as answer]
Yeah, you'd need a new bind group and bind group layout that store the image as a non-storage image, which you can then read on the shader using imageLoad.
